### ISSUE TITLE
Feature request: libc header import without libc library import

### ISSUE THREAD

**Issue by ityonemo** [ziglang/zig#7687]
For FFIs which use C ABI as a least common denominator, but I'm sure also some header-only C libraries, may not use any stdlib externs, but might still need accoutrements of the C stdlib headers (e.g. types.h).  It might be nice to provide a way to leverage zig's header resolution logic without doing `linkSystemLibrary("c")` which would unnecessarily bring in libc into the project.

**Comment by matu3ba** [comment]
Probably the documenation (#1596) should be finished first. Just curious: Is identifying manually the header files, which dont depend on libc, that difficult?

**Comment by ityonemo** [comment]
Yeah, it gets tricky (but not impossible) if you want cross-compilation to be possible.

**Comment by iacore** [comment]
As a workaround, use musl.

**Comment by matklad** [comment]
Appaerntly, if you don't _link_ to libc, but just have some `-I` entry, it works?

Here's my "C" library with a header file which doesn't really depend on libc at runtime:

```
λ bat mylib.h
#include <stdint.h>

int64_t get_number();

λ bat mylib.zig
export fn get_number() u64 {
    return 92;
}

λ zig build-lib mylib.zig
```

Here's the Zig code which imports the header

```
λ bat main.zig
const std = @import("std");
const c = @cImport(@cInclude("./mylib.h"));

pub fn main() void {
    std.debug.print("{}", .{c.get_number()});
}
```

If I just try to compile it, I  get the error I expect to get given this issue:

```
λ zig build-exe main.zig libmylib.a.o
main.zig:2:11: error: C import failed
const c = @cImport(@cInclude("./mylib.h"));
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
main.zig:2:11: note: libc headers not available; compilation does not link against libc
```

That is: no cImport if you don't link libc!

But, if I add `-I` for _my_ header, this somehow gets it to work!

```
λ zig build-exe main.zig libmylib.a.o -I. && ./main
92
λ ldd ./main
	not a dynamic executable
```

Notably, not only Zig is able to find my `mylib.h`, but internal clang is able to find `<stdint.h>`. looking at `--verbose-cimport`, it seems that it uses the correct stint.h:

```
λ bat /home/matklad/tmp/.zig-cache/o/e4d0a93094219ef8a9216d4c64d1b94a/cimport.h.d
cimport.o: \
  /home/matklad/tmp/.zig-cache/o/e4d0a93094219ef8a9216d4c64d1b94a/cimport.h \
  mylib.h /home/matklad/p/tb/work/zig/lib/include/stdint.h
```

So it looks like this might have been mostly fixed as of Zig 0.13, and only the error message needs to be adjusted: it's not about `libc` headers, it's about the parcitular header you try to `cImport`

**Comment by andrewrk** [comment]
Expected to be solved by #20630

**Comment by michaelxmroz** [comment]
Is there any other workaround for this issue? I am trying to use zig to compile C++ for a freestanding target, so I cannot link stdlib, but I would expect headers such as `<utility>` which are supposed to be [completely freestanding](https://eel.is/c++draft/utility.syn#header:%3cutility%3e) to be available. 

```
	const target = b.resolveTargetQuery(.{
        .abi = .eabihf,
        .cpu_arch = .aarch64,
        .os_tag = std.Target.Os.Tag.freestanding,
        .cpu_model = std.Target.Query.CpuModel{ .explicit = &std.Target.arm.cpu.cortex_a72 },
    });

	const optimize = b.standardOptimizeOption(.{});
	
	const static_lib = b.addStaticLibrary(.{ .name = "staticlib", .target = target, .optimize = optimize });
	static_lib.addIncludePath(b.path("."));
    static_lib.addCSourceFile(.{ .file = b.path("test.cpp"), .flags = &.{"-std=c++14", "-ffreestanding", "-o2", "-fbuiltin"} });
	static_lib.root_module.single_threaded = true;	
	b.installArtifact(static_lib);
```

Trying to build like this still results in an error though:

test.cpp:26:10: error: 'utility' file not found
#include <utility>
